modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/SlashApiHandler.java

```diff
@@ -213,12 +227,21 @@ private boolean verifyRequestMountedOnSites( final ApiDescriptor apiDescriptor,
             return false;
         }
 
-        final boolean isAppInstalledOnSite = site != null && site.getSiteConfigs().get( apiDescriptor.key().getApplicationKey() ) != null;
+        boolean isAppInstalledOnSite = site != null && site.getSiteConfigs().get( apiDescriptor.key().getApplicationKey() ) != null;
 
         if ( !isAppInstalledOnSite )
         {
-            Project project = projectService.get( ProjectName.from( portalRequest.getRepositoryId() ) );
-            return project != null && project.getSiteConfigs().get( apiDescriptor.key().getApplicationKey() ) != null;
+            final Project project = projectService.get( ProjectName.from( portalRequest.getRepositoryId() ) );
+            if ( project == null || project.getSiteConfigs().get( apiDescriptor.key().getApplicationKey() ) == null )
+            {
+                return false;
+            }
+        }
+
+        if ( apiDescriptor.getContextPath() == ApiContextPath.DEFAULT )
+        {
+            final String siteRelativePath = (String) ContextAccessor.current().getAttribute( SITE_RELATIVE_PATH_CTX_ATTRIBUTE );
```

Comment:
If we only use it in one place it is better to propagate it via method parameters